File: app/src/main/java/com/tt/handsomeman/di/module/SignUpModule.java
package com.tt.handsomeman.di.module;

import com.tt.handsomeman.service.SignUpService;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import retrofit2.Retrofit;

@Module
public class SignUpModule {

    @Provides
    @Singleton
    SignUpService provideSignUpService(Retrofit retrofit) {
        return retrofit.create(SignUpService.class);
    }
}

File: app/src/main/java/com/tt/handsomeman/di/module/SignUpModule.kt
package com.tt.handsomeman.di.module

import com.tt.handsomeman.service.SignUpService

import javax.inject.Singleton

import dagger.Module
import dagger.Provides
import retrofit2.Retrofit

@Module
class SignUpModule {

    @Provides
    @Singleton
    internal fun provideSignUpService(retrofit: Retrofit): SignUpService {
        return retrofit.create(SignUpService::class.java!!)
    }
}
